## Loading dataset from .csv

In [48]:
import pandas as pd

data_folder = 'data/'
itens_file_name = 'valor_unitario_aprovado_sample.csv'

itens_file_path = data_folder + itens_file_name

dt = pd.read_csv(itens_file_path)
dt.head()

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Unnamed: 0</th>
      <th>Item</th>
      <th>ValorUnitarioAprovado</th>
      <th>Data</th>
      <th>idPRONAC</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1894129</td>
      <td>Material de consumo</td>
      <td>7000.00</td>
      <td>2014-07-31 07:27:29</td>
      <td>179966</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2003098</td>
      <td>Locação de teatro</td>
      <td>465.39</td>
      <td>2015-05-05 16:38:24</td>
      <td>185736</td>
    </tr>
    <tr>
      <th>2</th>
      <td>925488</td>
      <td>Banheiro químico</td>
      <td>250.00</td>
      <td>2012-03-16 00:00:00</td>
      <td>143223</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1326505</td>
      <td>Técnico de som</td>
      <td>3500.00</td>
      <td>2013-01-07 00:00:00</td>
      <td>155368</td>
    </tr>
    <tr>
      <th>4</th>
      <td>136947</td>
      <td>Cartaz</td>
      <td>300.00</td>
      <td>2010-01-01 00:00:00</td>
      <td>114412</td>
    </tr>
  </tbody>
</table>
</div>

## Selecting a random item

When selecting a random item, it's important to make sure that we have enough occurences of that item, for example 100 or more. Let's also make a set (all itens are discinct) of random itens.

In [49]:
import numpy as np

def get_random_item():
    while True:
        item = np.random.choice(dt['Item'].values)
        if len(dt.loc[dt['Item'] == item]) >= 100:
            return item

def get_set_of_random_itens(n = 20):
    itens = set()
    while len(itens) < n:
        itens.add(get_random_item())
    return itens

print(get_set_of_random_itens())

{'Locação de teatro', 'Material de consumo', 'Registro videográfico', 'Banner/faixa adesiva/faixa de lona/saia de palco/testeira/pórtico\r\n', 'Produtor', 'INSS', 'Diretor de produção', 'Designer gráfico', 'Material de escritório', 'Folders', 'Assessor de imprensa', 'Mídia impressa', 'Contador', 'Direitos Autorais', 'Cartaz', 'Telefone', 'Correios', 'Coordenação geral', 'Hospedagem com Alimentação', 'Mídia radiofônica'}


### Getting the most frequent items


In [50]:
top_frequent = dt['Item'].value_counts().head(10)
display(top_frequent)

Transporte Local / Locação de Automóvel / Combustível               354
Passagens Aéreas (Descrever os trechos na tela de deslocamentos)    326
Refeição                                                            325
Assessor de imprensa                                                324
Assistente de produção                                              291
Produtor Executivo                                                  279
Contador                                                            261
Hospedagem sem Alimentação                                          243
Cartaz                                                              232
Remuneração para captação de recursos                               225
Name: Item, dtype: int64

### Number of distinct items

In [51]:
print(len(dt['Item'].unique()))

1106


**Lots of distinct items**

There 1106 distinct items in that sample. It will be hard to plot them one by one, so it's a good idea to plot some sample of those items

In [52]:
items_sample = dt['Item'].sample(10)
data_sample = dt[dt['Item'].isin(items_sample)]
data_sample_count = data_sample['Item'].value_counts()
display(data_sample.head(10))

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Unnamed: 0</th>
      <th>Item</th>
      <th>ValorUnitarioAprovado</th>
      <th>Data</th>
      <th>idPRONAC</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>18</th>
      <td>1316054</td>
      <td>Fotografia Artística (Fotógrafo, Tratamento, R...</td>
      <td>1500.0</td>
      <td>2013-04-01 00:00:00</td>
      <td>156618</td>
    </tr>
    <tr>
      <th>19</th>
      <td>823295</td>
      <td>Produtor</td>
      <td>800.0</td>
      <td>2012-01-10 00:00:00</td>
      <td>135716</td>
    </tr>
    <tr>
      <th>31</th>
      <td>353983</td>
      <td>Elenco Principal</td>
      <td>9100.0</td>
      <td>2010-05-17 00:00:00</td>
      <td>121581</td>
    </tr>
    <tr>
      <th>34</th>
      <td>605160</td>
      <td>Assessor de imprensa</td>
      <td>3500.0</td>
      <td>2012-01-01 00:00:00</td>
      <td>131173</td>
    </tr>
    <tr>
      <th>42</th>
      <td>309400</td>
      <td>Assessor de imprensa</td>
      <td>12000.0</td>
      <td>2010-05-30 00:00:00</td>
      <td>119919</td>
    </tr>
    <tr>
      <th>94</th>
      <td>1028230</td>
      <td>Assessor de imprensa</td>
      <td>4000.0</td>
      <td>2011-12-12 00:00:00</td>
      <td>145132</td>
    </tr>
    <tr>
      <th>111</th>
      <td>122064</td>
      <td>Produtor</td>
      <td>4000.0</td>
      <td>2010-10-01 00:00:00</td>
      <td>117277</td>
    </tr>
    <tr>
      <th>136</th>
      <td>2315055</td>
      <td>Assessor de imprensa</td>
      <td>3000.0</td>
      <td>2016-03-01 00:00:00</td>
      <td>199885</td>
    </tr>
    <tr>
      <th>157</th>
      <td>1690854</td>
      <td>Caixa de Produção</td>
      <td>600.0</td>
      <td>2014-03-15 00:00:00</td>
      <td>164527</td>
    </tr>
    <tr>
      <th>172</th>
      <td>78906</td>
      <td>Folder formato até A5(15,5x21cm)\r\n</td>
      <td>1800.0</td>
      <td>2009-07-01 00:00:00</td>
      <td>111811</td>
    </tr>
  </tbody>
</table>
</div>